### ISSUE TITLE
 ngIf not working with an @Input() value

### ISSUE THREAD

**Issue by rickblokdijk** [angular/angular#6318]
2.0.0-beta.0

`@Input() loading = false`
`loadingCheck = false`

HTML:

`<template [ngIf]="loading">`                -> does not hide template
`<template [ngIf]="loadingCheck">`       -> hides template

`{{loading}}` -> shows false

same issue when using *ngIf on an element

**Comment by zoechi** [comment]
`[ngIf]` should be `*ngIf`

**Comment by pkozlowski-opensource** [comment]
> [ngIf] should be *ngIf 

Not on a `<template>` element.

@rickblokdijk could you please provide a reproduce scenario using plunker? What you are describing sounds very odd.

**Comment by zoechi** [comment]
Sorry, my bad.

**Comment by pkozlowski-opensource** [comment]
I can **not** reproduce here: https://plnkr.co/edit/cg6EdMjnWJd4UUIswDVl?p=preview 

@rickblokdijk could you please modify this plunker to illustrate the issue you are describing? Otherwise it is not actionable.

**Comment by rickblokdijkvt** [comment]
@pkozlowski-opensource thanks for the plnkr, seems ok, here it's still not working, although I have some more complex structure. debugging my head of now, I will try to make a plnkr of the same situation. Will come back to it asap

**Comment by rickblokdijk** [comment]
Had to funnel it down to a simple example :). The loading value of the component is set through a service. While {{loading}} is false, it does not hide the elements. See:

https://plnkr.co/edit/fY4DdfBOz9carkD9DCU3?p=preview

**Comment by pkozlowski-opensource** [comment]
TL;DR; Everything works as expected, here is a "fixed" plunk: https://plnkr.co/edit/aqFkr5QhJWMuCLCQ5uMS?p=preview

@rickblokdijk you might be miss-interpreting how bindings work in Angular 2. If you do `loading="{{appService.loading}}"` it is equivalent of doing `[loading]="'' + appService.loading"` (stringifing a value).

`'' + false` gives `"false"` which is truthy in JavaScript...

**Comment by rickblokdijk** [comment]
Clear, understand now. Sorry for your time, good luck!
